edb/server/compiler/compiler.py

```diff
@@ -537,6 +537,76 @@ def _compile_ql_script(
 
         return sql.decode(), argmap
 
+    def _compile_query_args(
+            self,
+            ctx: CompileContext,
+            schema: s_schema.Schema,
+            subtypes: List[Tuple[str, Any, bool]],
+    ) -> Tuple[bytes, uuid.UUID]:
+        current_tx = ctx.state.current_tx()
+
+        if not subtypes:
+            arg_ast = qlast.SelectQuery(
+                orderby=[],
+                result=qlast.Path(steps=[qlast.ObjectRef(
+                    name='FreeObject',
+                    module='std',
+                )]),
+            )
+        else:
+            elements = []
+            for name, schema_type, required in subtypes:
+                if required:
+                    card = qlast.CardinalityModifier.Required
+                else:
+                    card = qlast.CardinalityModifier.Optional
+
+                elem = qlast.ShapeElement(
+                    expr=qlast.Path(steps=[qlast.Ptr(
+                        ptr=qlast.ObjectRef(name=f'__arg_{name}'),
+                        direction=s_pointers.PointerDirection.Outbound,
+                    )]),
+                    compexpr=qlast.TypeCast(
+                        cardinality_mod=card,
+                        optional=not required,
```

Comment:
`TypeCast` doesn't have an `optional` attribute.  I'm surprised `mypy` didn't complain here, since our mypy plugin is injecting strict `__init__` signatures.  @msullivan, ideas?

Reply:
changed